### ISSUE TITLE
Azure LB becomes unconfigurable if a single NIC part of the backend pool Fails

### ISSUE THREAD

**Issue by djsly** [kubernetes/kubernetes#65025]
**Is this a BUG REPORT or FEATURE REQUEST?**:

> Uncomment only one, leave it on its own line:
>
/kind bug
> /kind feature

**What happened**:
We are trying to create Kubernetes Services of type LoadBalancer, the Load Balancer cannot accept Write/Delete operation and it is complaining about a Failed NIC.

```
Cannot proceed with operation because resource /subscriptions/<subID>/resourceGroups/cne2-test/providers/Microsoft.Network/networkInterfaces/kubenode3_ni_default/ipConfigurations/ipconf1 used by resource /subscriptions/<subID>/resourceGroups/cne2-test/providers/Microsoft.Network/loadBalancers/cne2-test is not in Succeeded state. Resource is in Failed state and the last operation that updated/is updating the resource is PutNicOperation.
```
All operation on this LB is throwing this error out. 

**What you expected to happen**:

That the Service IP gets provisioned in the LB, and a Failed NIC shouldn't affect the creation of a service LB like that. 

**How to reproduce it (as minimally and precisely as possible)**:
No Clue how to get a NIC into a Failed State, that should do it though. 

**Anything else we need to know?**:

**Environment**:
- Kubernetes version (use `kubectl version`):
```
Client Version: version.Info{Major:"1", Minor:"10", GitVersion:"v1.10.2", GitCommit:"81753b10df112992bf51bbc2c2f85208aad78335", GitTreeState:"clean", BuildDate:"2018-04-27T09:22:21Z", GoVersion:"go1.9.3", Compiler:"gc", Platform:"linux/amd64"}
Server Version: version.Info{Major:"1", Minor:"10", GitVersion:"v1.10.2", GitCommit:"81753b10df112992bf51bbc2c2f85208aad78335", GitTreeState:"clean", BuildDate:"2018-04-27T09:10:24Z", GoVersion:"go1.9.3", Compiler:"gc", Platform:"linux/amd64"}
```
- Cloud provider or hardware configuration:
```
Azure China East2
```
- OS (e.g. from /etc/os-release):
```
NAME="CentOS Linux"
VERSION="7 (Core)"
ID="centos"
ID_LIKE="rhel fedora"
VERSION_ID="7"
PRETTY_NAME="CentOS Linux 7 (Core)"
ANSI_COLOR="0;31"
CPE_NAME="cpe:/o:centos:centos:7"
HOME_URL="https://www.centos.org/"
BUG_REPORT_URL="https://bugs.centos.org/"

CENTOS_MANTISBT_PROJECT="CentOS-7"
CENTOS_MANTISBT_PROJECT_VERSION="7"
REDHAT_SUPPORT_PRODUCT="centos"
REDHAT_SUPPORT_PRODUCT_VERSION="7"
```
- Kernel (e.g. `uname -a`):
```
Linux 3.10.0-693.21.1.el7.x86_64 #1 SMP Wed Mar 7 19:03:37 UTC 2018 x86_64 x86_64 x86_64 GNU/Linux
```
- Install tools:
InHouse, Upstream + CloudProvider Logic
- Others:
CC : @feiskyer

**Comment by djsly** [comment]
@kubernetes/sig-azure-bugs

**Comment by feiskyer** [comment]
@djsly This is a limitation of Azure API, and the failed state is not taken into consideration yet in controller manager. We should definitely handle this in Azure could provider.

Meanwhile, a simple way to mitigate the issue is fix nic state, e.g.

```
az network nic update --resource-group <resource-group> --name <nic-name>
```

**Comment by brendandburns** [comment]
cc @khenidak

**Comment by djsly** [comment]
@feiskyer removing the Failed status on the NIC did allow the LB to start accepting write/delete requests, thanks. 

Keeping open to track the updated controller-manager logic.

**Comment by feiskyer** [comment]
Fixed in #65412.
